### ISSUE TITLE
BUFR data request

### ISSUE THREAD

**Issue by synoptic550** [jweyn/MOS-X#1]
Hi Weyn，

You mentioned BUFR data were available beginning 2010-01-01, however, I see only the latest 7 days' data are available using bufr_gruven.pl.  Could you please tell me how to get the historical model data? Thanks!

Here is the log for requsting BUFR data.
I.  Determining which BUFR files need to be acquired

           Initiating HTTP connection to ems2.comet.ucar.edu

             -> Checking if available /data/bufr/201002/20100202/gfs/bufr.t18z/bufr3.727930.2010020218  - Not currently available

           Initiating HTTP connection to ems1.comet.ucar.edu

             -> Checking if available /data/bufr/201002/20100202/gfs/bufr.t18z/bufr3.727930.2010020218  - Not currently available

           Initiating HTTP connection to ems3.comet.ucar.edu

             -> Checking if available /data/bufr/201002/20100202/gfs/bufr.t18z/bufr3.727930.2010020218  - Not currently available

         Sorry about your BUFR files. I'll do a better job next time!

BR,
Yangyang Zhao

**Comment by yeechianlow** [comment]
Hi Yangyang,

I'm not Weyn, but the link "/data/bufr/201002/20100202/gfs/bufr.t18z/bufr3.727930.2010020218" is available for me. However, it seems like you have to call bufr_gruven.pl with gfs3, not gfs. Also, for NAM, you have to uncomment the "SERVER-HTTP = EMS3:/data/bufr/YYYYMM/YYYYMMDD/nam/bufr.tCCz/bufr.STNM.YYYYMMDDCC" line in the nam_bufrinfo.conf file, as that is the only link that has data going back to 2010. The NOAA sites only keep the last 7 days.

Yeechian

**Comment by jweyn** [comment]
Hi @synoptic550, are you having any issues with the MOS-X code retrieving the bufr files? I think @yeechianlow is right (use `gfs3`, although MOS-X should do that automatically). My version of bufrgruven (17.25) has all the servers uncommented in the .conf files, so I recommend checking that.

Apologies for the delay, I never got any notifications.

**Comment by joejoezz** [comment]
There are some additional issues with the latest release of bufrgruven (19.24). The NAM uncommenting mentioned by @yeechianlow successfully gets the NAM to work. However, the GFS has some new problems related to `gfs3` being no longer supported. After a ton of hair pulling I figured out that the EMS3 server changed its GFS bufr file names in mid-2017:

- pre-August 2017: `bufr3.STNM.YYYYMMDDCC`
- post-August 2017: `bufr.STNM.YYYYMMDDCC`

As of right now I do not have a good fix for this...one can adjust `gfs_bufrinfo.conf` to find the <bufr3> files but the way bufrgruven is written it will not read duplicate server URLs in bufrgruven.conf. There is probably a way to modify the HTTP subroutine of Method.pm with a wildcard character to handle `bufr` or `bufr3`. For now I have created an ugly hack to `bufrgruven.conf` with an "EMS4" server as follows:

```
EMS3     = ems3.comet.ucar.edu
EMS4     = ems3.comet.ucar.edu/ 
```
and changed `gfs_bufrinfo.conf` as follows:
```
SERVER-HTTP = EMS3:/data/bufr/YYYYMM/YYYYMMDD/gfs/bufr.tCCz/bufr3.STNM.YYYYMMDDCC
SERVER-HTTP = EMS4:data/bufr/YYYYMM/YYYYMMDD/gfs/bufr.tCCz/bufr.STNM.YYYYMMDDCC

```
